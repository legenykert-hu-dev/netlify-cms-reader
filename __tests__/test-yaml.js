const { getConfig, getData } = require('../index')

test('reads yaml data', async () => {
  const config = getConfig('__tests__/yaml/config.yml')
  const data = await getData(config)
  expect(data).toMatchSnapshot()
})

test('reads yaml data with filtered folder collections', async () => {
  const config = getConfig(
    '__tests__/yaml/config-with-filtered-folder-collections.yml'
  )
  const data = await getData(config)
  expect(data).toMatchSnapshot()
})

test('honors first of the sortableField list', async () => {
  const config = getConfig(
    '__tests__/yaml/config-with-sortableField-in-folder-collections.yml'
  )
  const data = await getData(config)
  expect(data).toMatchSnapshot()
})
