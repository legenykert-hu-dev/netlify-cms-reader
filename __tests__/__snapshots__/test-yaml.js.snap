// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`honors first of the sortableField list 1`] = `
Object {
  "sortable": Array [
    Object {
      "content": "# Page With Index 1 Content",
      "index": 1,
      "title": "Page With Index 1",
    },
    Object {
      "content": "# Page With Index 2 Content",
      "index": 2,
      "title": "Page With Index 2",
    },
  ],
}
`;

exports[`reads yaml data 1`] = `
Object {
  "layout": Object {
    "footer": Object {
      "content": "Footer Content",
      "copyright": "Â© 2019 Copyright",
    },
    "navbar": Object {
      "title": "Navbar Title",
    },
  },
  "pages": Array [
    Object {
      "content": "# About Page Content",
      "path": "/about",
      "title": "About Page",
    },
    Object {
      "content": "# Index Page Content",
      "path": "/",
      "title": "Index Page",
    },
  ],
}
`;

exports[`reads yaml data with filtered folder collections 1`] = `
Object {
  "filtered_to_a_and_b": Array [
    Object {
      "content": "# Content with filter set to A or B",
      "filter": Array [
        "A",
        "B",
      ],
      "title": "To be filtered to A or B",
    },
  ],
  "filtered_to_c_and_d": Array [
    Object {
      "content": "# Content with filter set to C or D",
      "filter": Array [
        "D",
        "C",
      ],
      "title": "To be filtered to C or D",
    },
  ],
  "unfiltered": Array [
    Object {
      "content": "# Content with filter set to A or B",
      "filter": Array [
        "A",
        "B",
      ],
      "title": "To be filtered to A or B",
    },
    Object {
      "content": "# Content with filter set to C or D",
      "filter": Array [
        "D",
        "C",
      ],
      "title": "To be filtered to C or D",
    },
  ],
}
`;
